examples/templates/cluster-operator/ephemeral-template.yaml

```diff
@@ -63,6 +63,10 @@ parameters:
   displayName: Transaction state replication factor
   name: KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR
   value: "3"
+- description: The version of the format using for storing messages
+  displayName: Log message format version
+  name: KAFKA_LOG_MESSAGE_FORMAT_VERSION
+  value: "2.1"
```

Comment:
Does this really deserve to be a parameter of the template? Wouldn't it be enough to set `log.message.format.version: ${{KAFKA_VERSION}}`

Reply:
The `KAFKA_VERSION` is `2.1.0` while message format `2.1`. Do you know if the broker is fine setting the message format to `2.1.0` (so using the `patch` digit as well)?

Reply:
Good point. I _think_ it will work, but you'll need to check the Kafka sources to see how it's handled, I guess.

Reply:
I first did a running test and it works then I took a look at the Kafka code and it seems that the main points are the following:

`val logMessageFormatVersion = ApiVersion(logMessageFormatVersionString)`

where `ApiVersion` has the `apply` in this way:

```
def apply(versionString: String): ApiVersion = {
    val versionSegments = versionString.split('.').toSeq
    val numSegments = if (versionString.startsWith("0.")) 3 else 2
    val key = versionSegments.take(numSegments).mkString(".")
    versionMap.getOrElse(key, throw new IllegalArgumentException(s"Version `$versionString` is not a valid version"))
  }
```

which means that "2.1.0" is transformed in just "2.1" so it works.

Btw I was thinking the possibility that for a certain amount of time, the user would want to have old message format version (i.e. 2.0) but with new Kafka version (i.e. 2.1) so we should need two different template parameters as the current PR. @tombentley wdyt?

Reply:
> for a certain amount of time, the user would want to have old message format version (i.e. 2.0) but with new Kafka version (i.e. 2.1)

Why would they want to do that (given the template is presumably being used to create a _new_ cluster)?

Reply:
The new Kafka version could have a bug fix unrelated to the clients and message formats so that the user would still use clients with old message format (without upgrading them) but having the newer Kafka version with some bug fixes?

Reply:
That seems really tenuous to me. My understanding is that using a broker with a non-default message format imposes a perf penalty which means that there's usually a strong incentive to use the default. The templates are really supposed to offer an easy way of creating the resource. The more parameters there are the harder it is for the user to understand them enough to decide whether they should change them. So for me, on balance, I think the simplicity of not having a parameter for this is to be preferred.

Reply:
Why is it so "tenuous"? How many problems can have a customer to update all their clients when it's not needed while he wants to have latest Kafka version for fixing a bug on an unrelated stuff. 
When things work fine, customers don't want to update software. At least it's my experience (a lot of customers run Windows CE 5.0 devices even after 10-15 years when new versions came out).

Reply:
My point is that this is something of a corner case for which you're prepared to make the usual case more complex. I'm not preventing users who want to do what you describe from doing it, if they want: They can always create the resource themselves without using the template. Simple things should be simple, hard things should be possible and all that.